doc/Customization-guide/Endpoint-SDKs/C-SDK/Porting-guide/index.md

```diff
@@ -4,3 +4,67 @@ title: Porting guide
+{:toc}
+
+
+# Prerequisites
+
+Kaa C SDK can be ported to most platforms with OS, without any OS and it's "endian safe".
+
+> **NOTE:**  It supports only 32-bits and 64-bits platforms. A device needs to have a network controller with TCP-IP stack.
+
+# Dependencies
+
+The SDK uses a platform-level layer that depends on the environment where the SDK should run. 
+If you want to port it, you need to modify a few files of the SDK and implement platform-dependent functions. These files can be divided into the next categories:
+
+## Mandatory             
+    
+ - Dynamic memory allocation, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/mem.h). 
+
+ - Networking stack, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/sock.h).
+
+ - Time utilities, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/time.h).
+ 
+ - Output stream, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/stdio.h).
+ 
+ - Default settings, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/defaults.h).
+  
+> **NOTE:** All references are files which are used by esp8266. You need to create the same but with own implementations.
+ 
+## Optional         
+
+ - Compile time generated and run time generated keys of the encryption. For example, the ESP8266 demo uses only the compile time generated key.
+
+ - SHA-1 calculation (you may use a default implementation)
+ 
+ > **NOTE:** If you want to save memory and the performance of a device, better to use own implementation of this code. For example, it will good for devices which have a small size of the data (32 kB) and code memory (128 kB).
+  
+## Additional
+
+ - Log and debug.  
+ 
+ - Encryption. It supports to disable or enable the encryption.
+    
+> **NOTE:** For small platforms (microcontrollers) we recommend to use the encryption with the compile time generated key, because the run time generated key doesn't give a big dispersion and it will save the resource of the processor.
+More information about these dependencies you can find in a [file]({{github_url}}client/client-multi/client-c/CMakeLists.txt).
+  
+  
+# Examples
+   
+[The code]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl) which was implemented for a few platforms can help, if you want to implement own platform.                        
+[The custom app]({{root_url}}Programming-guide/Your-first-Kaa-application/) can help to see how platform-dependent code is working and then explore existing [implementation]({{github_url}}client/client-multi/client-c ) for our C SDK.
+
+# CMake files
+
+There are two files:
+
+ - A CMakelists.txt file of the platform. The file has patches to the files of the platform. You need to implement own file and add it to a [directory]({{github_url}}client/client-multi/client-c/listfiles/platform).
```

Comment:
What means `The file has patches to the files of the platform`? What is a name convention for these type of files? Purpose is absolutely unclear, please specify.